api/v1alpha3/vspherevm_webhook.go

```diff
@@ -35,6 +36,19 @@ func (r *VSphereVM) SetupWebhookWithManager(mgr ctrl.Manager) error {
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-vspherevm,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=vspherevms,versions=v1alpha3,name=validation.vspherevm.infrastructure.x-k8s.io,sideEffects=None
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha3-vspherevm,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=vspherevms,versions=v1alpha3,name=default.vspherevm.infrastructure.x-k8s.io,sideEffects=None
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *VSphereVM) Default() {
+	// Windows hostnames must be < 16 characters in length
+	if r.Spec.OS == Windows && len(r.Name) > 15 {
+		name, err := base36TruncatedHash(r.Name, 15)
```

Comment:
just thinking, for UX, could we have the last 5 chars be random, and the first part == the windows VM name so theyre easy to read?

Reply:
yeah that could work. we currently use the machine deployment name `<cluster-name>-windows` where the default linux machine deployment is just `<cluster-name>`. With just 15 characters, it's hard to fit something meaningful that would reflect the cluster it belongs to and differentiate Windows while staying unique

Reply:
Did the NETBIOS limitation turn out to be important?